security.go

```diff
@@ -0,0 +1,50 @@
+package slack
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+// SecretsVerifier contains the information needed to verify that the request comes from Slack
+type SecretsVerifier struct {
+	slackSig    string
+	timeStamp   string
+	requestBody string
+}
+
+// NewSecretsVerifierFromHeader returns a new SecretsVerifier object in exchange for an http.Header object
+func NewSecretsVerifierFromHeader(header http.Header) (SecretsVerifier, error) {
+	if header["X-Slack-Signature"][0] == "" || header["X-Slack-Request-Timestamp"][0] == "" {
+		return SecretsVerifier{}, errors.New("headers are empty, cannot create SecretsVerifier")
+	}
+
+	return SecretsVerifier{
+		slackSig:  header["X-Slack-Signature"][0],
+		timeStamp: header["X-Slack-Request-Timestamp"][0],

```

Comment:
pass the secret through the intializer. 

should be
```golang
return SecretsVerifier{
	slackSig:  header["X-Slack-Signature"][0],
	timeStamp: header["X-Slack-Request-Timestamp"][0],
        hmac: hmac.New(sha256.New, signingSecret),
}
```

Reply:
the signing secret is an environment variable obtained from the admin panel--it isn't contained in the header so are you suggesting passing it in as an argument? also it is a string so will need to be converted to bytes to pass in to `hmac.New`

Reply:
i'm psuedo coding for you, not ensuring everything is exactly correct.

Reply:
understood-- thanks and sorry. just wanted to confirm we were on the same page about the meaning of the `signingSecret`